_examples/digitalidentity/main.go

```diff
@@ -0,0 +1,167 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"html/template"
+	"log"
+	"net/http"
+	"os"
+	"path"
+
+	"github.com/getyoti/yoti-go-sdk/v3"
+	"github.com/getyoti/yoti-go-sdk/v3/digitalidentity"
+	_ "github.com/joho/godotenv/autoload"
+)
+
+type contextKey string
+
+var (
+	errApplyingTheParsedTemplate = "Error applying the parsed template: "
+	errParsingTheTemplate        = "Error parsing the template: "
+)
+
+func home(w http.ResponseWriter, req *http.Request) {
+	templateVars := map[string]interface{}{
+		"yotiScenarioID":  os.Getenv("YOTI_SCENARIO_ID"),
+		"yotiClientSdkID": os.Getenv("YOTI_CLIENT_SDK_ID")}
+	t, err := template.ParseFiles("login.html")
+
+	if err != nil {
+		errorPage(w, req.WithContext(context.WithValue(
+			req.Context(),
+			contextKey("yotiError"),
+			fmt.Sprintf(errParsingTheTemplate+err.Error()),
+		)))
+		return
+	}
+
+	err = t.Execute(w, templateVars)
+	if err != nil {
+		errorPage(w, req.WithContext(context.WithValue(
+			req.Context(),
+			contextKey("yotiError"),
+			fmt.Sprintf(errApplyingTheParsedTemplate+err.Error()),
+		)))
+		return
+	}
+}
+func buildDigitalIdentitySessionReq() (sessionSpec *digitalidentity.ShareSessionRequest, err error) {
+	policy, err := (&digitalidentity.PolicyBuilder{}).WithFullName().WithEmail().WithPhoneNumber().WithSelfie().WithAgeOver(18).WithNationality().WithGender().WithDocumentDetails().WithDocumentImages().WithWantedRememberMe().Build()
+	if err != nil {
+		return nil, fmt.Errorf("failed to build policy: %v", err)
+	}
+
+	subject := []byte(`{
+		"subject_id": "unique-user-id-for-examples"
+	}`)
+
+	sessionReq, err := (&digitalidentity.ShareSessionRequestBuilder{}).WithPolicy(policy).WithRedirectUri("https:/www.yoti.com").WithSubject(subject).Build()
+	if err != nil {
+		return nil, fmt.Errorf("failed to build create session request: %v", err)
+	}
+	return &sessionReq, nil
+}
+
+func generateSession(w http.ResponseWriter, r *http.Request) {
+	didClient, err := initialiseDigitalIdentityClient()
+	if err != nil {
+		fmt.Fprintf(w, string("Client could't be generated"))
```

Comment:
missing return

Reply:
This method doesnt have return so we just returning result with fmt.Fprintf

Reply:
even if a function doesn't have a return value, you still have to `return` from the function after this error since we cannot continue